src/libsystemd/sd-rtnl/sd-rtnl.c

```diff
@@ -1006,13 +984,39 @@ int sd_rtnl_add_match(sd_rtnl *rtnl,
                       sd_rtnl_message_handler_t callback,
                       void *userdata) {
         struct match_callback *c;
+        int r;
 
         assert_return(rtnl, -EINVAL);
         assert_return(callback, -EINVAL);
         assert_return(!rtnl_pid_changed(rtnl), -ECHILD);
-        assert_return(rtnl_message_type_is_link(type) ||
-                      rtnl_message_type_is_addr(type) ||
-                      rtnl_message_type_is_route(type), -EOPNOTSUPP);
+
+        /* TODO: support all broadcast groups, and also leave them again when we no
+                 longer care */
```

Comment:
How about a simple ref-count per type? This way you'll leave the group again properly. And we can drop the TODO item then.